pepys_import/resolvers/command_line_resolver.py

```diff
@@ -107,56 +121,148 @@ def resolve_sensor(self, data_store, sensor_name, sensor_type, host_id, privacy,
+            options,
+            validate_method=is_valid_dynamic,
+        )
+        if choice == ".":
+            print("-" * 61, "\nReturning to the previous menu\n")
+            return None
+        elif choice == str(1):
+            result = self.fuzzy_search_reference(
+                data_store, change_id, data_type, db_class, field_name, text_name
+            )
             if result is None:
-                return self.resolve_privacy(data_store, change_id, data_type)
+                return self.resolve_reference(
+                    data_store, change_id, data_type, db_class, field_name, text_name
+                )
             else:
                 return result
         elif choice == str(2):
-            new_privacy = prompt("Please type name of new classification: ")
-            privacy = data_store.search_privacy(new_privacy)
-            if privacy:
-                return privacy
-            elif new_privacy:
-                return data_store.add_to_privacies(new_privacy, change_id)
+            new_object = prompt(f"Please type name of new {text_name}: ")
+            search_method = getattr(data_store, f"search_{field_name}")
+            obj = search_method(new_object)
+            if obj:
+                return obj
+            elif new_object:
+                add_method = getattr(data_store, f"add_to_{plural_field}")
+                return add_method(new_object, change_id)
             else:
                 print("You haven't entered an input!")
-                return self.resolve_privacy(data_store, change_id, data_type)
-        elif choice == ".":
+                return self.resolve_reference(
+                    data_store, change_id, data_type, db_class, field_name, text_name
+                )
+        elif 3 <= int(choice) <= len(options):
+            selected_object = objects_dict[options[int(choice) - 1]]
+            if selected_object:
+                return selected_object
+
+    def fuzzy_search_reference(
+        self, data_store, change_id, data_type, db_class, field_name, text_name=None
+    ):
+
+        objects = data_store.session.query(db_class).all()
+        completer = [p.name for p in objects]
+        choice = create_menu(
+            "Please start typing to show suggested values",
+            cancel=f"{text_name} search",
+            choices=[],
+            completer=FuzzyWordCompleter(completer),
+        )
+        if choice == ".":
             print("-" * 61, "\nReturning to the previous menu\n")
             return None
+        elif choice not in completer:
+            new_choice = create_menu(
+                f"You didn't select an existing {text_name}. " f"Do you want to add '{choice}' ?",
+                choices=["Yes", f"No, I'd like to select an existing {text_name}"],
+                validate_method=is_valid,
+            )
+            if new_choice == str(1):
+                plural_field = (
+                    re.sub(
```

Comment:
Could you add a comment on what this regexp is aiming to do?

Reply:
Added:
https://github.com/debrief/pepys-import/pull/408/commits/887fdf8132ae3643f9be5fe7892f2fba5b66a097